### PR TITLE
State based routing and navigation

### PR DESCRIPTION
**Pull request by Michael-R-Mac**
Hello, Part1 is okay, the movie card and movie view are okay. The part2 should be improved. I am trying to do the same has with movie view to show the user information. In order to update I am using a similar approach to signup and the same for delete.

### CODE REVIEW

File: src/components/MainView/MainView.jsx [hunk 1]
```diff
@@ -3,19 +3,23 @@ import { MovieCard } from "../MovieCard/MovieCard";
 import { MovieView } from "../MovieView/MovieView";
 import { LoginView } from "../LoginView/LoginView";
 import { SignupView } from "../SignUpView/SignUpView";
+import { NavigationBar } from "../NavigationBar/NavigationBar";
+import { ProfileView } from "../ProfileView/ProfileView";
+import Row from "react-bootstrap/Row";
```
Comment (frankhn) [inline src/components/MainView/MainView.jsx]: Make use of distructaring 

import {Row, Col} from 'react-bootstrap';

File: src/components/MainView/MainView.jsx [hunk 2]
```diff
@@ -3,19 +3,23 @@ import { MovieCard } from "../MovieCard/MovieCard";
...
+import Col from "react-bootstrap/Col";
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
 
 export const MainView = () => {
   const storedUser = JSON.parse(localStorage.getItem("user"));
   const storedToken = localStorage.getItem("token");
-  const [user, setUser] = useState(storedUser? storedUser : null);
-  const [token, setToken] = useState(storedToken? storedToken : null);
+  const [user, setUser] = useState(storedUser ? storedUser : null);
```
Comment (frankhn) [inline src/components/MainView/MainView.jsx:14]: Not best practice, checkout react hooks useEffect or componentWillUpdate

File: src/components/MainView/MainView.jsx [hunk 3]
```diff
@@ -26,50 +30,149 @@ export const MainView = () => {
...
-        }} />
-        or
-        <SignupView />
-      </>
-    );
-  }
-
-  if (selectedMovie) {
-    return (
-      <><button onClick={() => { setUser(null); setToken(null); localStorage.clear(); }}>Logout</button>
```
Comment (frankhn) [inline src/components/MainView/MainView.jsx]: Better to have a separate function for this

File: src/components/MainView/MainView.jsx [hunk 4]
```diff
@@ -26,50 +30,149 @@ export const MainView = () => {
...
-            setSelectedMovie(newSelectedMovie);
-          } } />
-      ))}
-    </div></>
+    <BrowserRouter>
+      <NavigationBar
+        user={user}
+        onLoggedOut={() => {
+          setUser(null);
+          setToken(null);
```
Comment (frankhn) [inline src/components/MainView/MainView.jsx:69]: Same as the above, separation of concern

File: src/components/MainView/MainView.jsx [hunk 5]
```diff
@@ -26,50 +30,149 @@ export const MainView = () => {
...
+                        <Col className="mb-4" key={movie.id} md={3}>
+                          <MovieCard movie={movie} />
+                        </Col>
+                      ))}
+                    </>
+                    <button
+                      onClick={() => {
+                        setUser(null);
+                        setToken(null);
+                        localStorage.clear();
```
Comment (frankhn) [inline src/components/MainView/MainView.jsx:152]: Here as well

**Review by frankhn — CHANGES_REQUESTED** [review]
Everything else looks ok,

The changes above are crucial for a world developer like you Mic

File: src/components/MainView/MainView.jsx (hunk 1 above)
Comment (Michael-R-Mac) [reply src/components/MainView/MainView.jsx]: Thank you. Changed.

Comment (frankhn) [reply src/components/MainView/MainView.jsx]: You have conflicts

Comment (Michael-R-Mac) [reply src/components/MainView/MainView.jsx]: I am still doing your corrections. By conflicts you mean the next comment?

Comment (frankhn) [reply src/components/MainView/MainView.jsx]: I mean Branch conflict with the target branch
image

File: src/components/MainView/MainView.jsx (hunk 2 above)
Comment (Michael-R-Mac) [reply src/components/MainView/MainView.jsx:14]: In exercise 3.5 that practice is recommended.

File: src/components/MainView/MainView.jsx (hunk 3 above)
Comment (Michael-R-Mac) [reply src/components/MainView/MainView.jsx]: Red means erased.

File: src/components/MainView/MainView.jsx (hunk 5 above)
Comment (Michael-R-Mac) [reply src/components/MainView/MainView.jsx:152]: Again refer to exercise 3.5, that is recommended.

File: src/components/MainView/MainView.jsx (hunk 4 above)
Comment (Michael-R-Mac) [reply src/components/MainView/MainView.jsx:69]: In exercise 3.7 in the sandbox that is recommended.

**Pull request merged**